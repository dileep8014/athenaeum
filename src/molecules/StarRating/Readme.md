### StarRating Example:
Small StarRating

```jsx
  <StarRating size={12} rating={5} />
```
Large StarRating

```jsx
  <StarRating size={18} rating={4} />
```
