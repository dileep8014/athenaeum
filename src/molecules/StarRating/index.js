import React from 'react'; import PropTypes from 'prop-types';
import classnames from 'classnames';
import Icon from 'atoms/Icon';
import styles from './star_rating.module.scss';

function StarRating(props) {
  const {
    className,
    size,
    rating,
  } = props;

  const classes = classnames(
    styles['star-rating'],
    styles.stars,
    styles[`stars-${rating}`],
    className,
  );

  return (
    <span className={classes}>
      <Icon
        className={classnames(
          styles.star,
          {
            [styles['filled-in']]: rating > 0
          }
        )}
        icon='star2'
        width={size}
        height={size}
        renderSVGDOM
      />
      <Icon
        className={classnames(
          styles.star,
          {
            [styles['filled-in']]: rating > 1
          }
        )}
        icon='star2'
        width={size}
        height={size}
        renderSVGDOM
      />
      <Icon
        className={classnames(
          styles.star,
          {
            [styles['filled-in']]: rating > 2
          }
        )}
        icon='star2'
        width={size}
        height={size}
        renderSVGDOM
      />
      <Icon
        className={classnames(
          styles.star,
          {
            [styles['filled-in']]: rating > 3
          }
        )}
        icon='star2'
        width={size}
        height={size}
        renderSVGDOM
      />
      <Icon
        className={classnames(
          styles.star,
          {
            [styles['filled-in']]: rating > 4
          }
        )}
        icon='star2'
        width={size}
        height={size}
        renderSVGDOM
      />
    </span>
  );
}


StarRating.propTypes = {
  /**
   * This prop will add a new className to any inherent classNames
   * provided in the component's index.js file.
   */
  className: PropTypes.string,

  /**
   * Add custom inline style
   */
  style: PropTypes.object,

  /**
   * Pixel amount for height and width of stars.
   */
  size: PropTypes.string,

  /**
   * Number of stars that are filled in.
   */
  rating: PropTypes.number.isRequired,
};

StarRating.defaultProps = {
  height: '12px',
  // Place any default props here.
};

export default StarRating;
