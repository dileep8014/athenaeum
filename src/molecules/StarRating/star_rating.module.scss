.star-rating {
  // Place your component specific scss here.
  line-height: 0;
}

.star {
  margin-left: rem-calc(6.4px);

  &:first-of-type {
    margin-left: 0;
  }

  svg polygon {
    fill: color('neutral-4');
    stroke: color('neutral-4');
  }
}

.filled-in {
  svg polygon {
    fill: color('primary-3');
    stroke: color('primary-3');
  }
}
