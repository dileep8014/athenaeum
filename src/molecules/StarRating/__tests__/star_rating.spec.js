import React from 'react';
import renderer from 'react-test-renderer';

import StarRating from 'molecules/StarRating';

describe('<StarRating />', () => {
  beforeEach(() => {
    global.fetch = () => new Promise((resolve) => resolve());
  });

  it('renders correctly', () => {
    const actual = renderer.create(<StarRating rating={4} />).toJSON();

    expect(actual).toMatchSnapshot();
  });
});
