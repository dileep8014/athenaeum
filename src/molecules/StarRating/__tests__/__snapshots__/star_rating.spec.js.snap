// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StarRating /> renders correctly 1`] = `
<span
  className="star-rating stars stars-undefined"
>
  <span
    className="icon-wrapper star"
    onClick={undefined}
    style={
      Object {
        "height": undefined,
        "width": undefined,
      }
    }
  />
  <span
    className="icon-wrapper star"
    onClick={undefined}
    style={
      Object {
        "height": undefined,
        "width": undefined,
      }
    }
  />
  <span
    className="icon-wrapper star"
    onClick={undefined}
    style={
      Object {
        "height": undefined,
        "width": undefined,
      }
    }
  />
  <span
    className="icon-wrapper star"
    onClick={undefined}
    style={
      Object {
        "height": undefined,
        "width": undefined,
      }
    }
  />
  <span
    className="icon-wrapper star"
    onClick={undefined}
    style={
      Object {
        "height": undefined,
        "width": undefined,
      }
    }
  />
</span>
`;
